src/provider/model/ClientHandler.ts

```diff
@@ -0,0 +1,93 @@
+import {Identity} from 'openfin/_v2/main';
+import {ApplicationOption} from 'openfin/_v2/api/application/applicationOption';
+import {injectable, inject} from 'inversify';
+import {Signal} from 'openfin-service-signal';
+
+import {APITopic, Events} from '../../client/internal';
+import {Inject} from '../common/Injectables';
+import {Injector} from '../common/Injector';
+
+import {APIHandler} from './APIHandler';
+import {CollectionMap} from './database/Database';
+
+export type AppInitData = {
+    id: string;
+    data: ApplicationOption | string
+}
+
+/**
+ * Client handler is responsible of keeping track of active clients, storing information about them
+ * and re-launching them when requested by other modules. Is intended to be used solely by EventPump.
+ *
+ */
+@injectable()
+export class ClientHandler {
```

Comment:
Can we review the naming of this. It has a pretty specific purpose but a rather generic name, that also rather overlaps with what `APIHandler` does (as that's the part of the provider that manages all client communication).

This also seems rather bound to `EventPump`, as the comment above also states. Could the purpose be solidified by merging these together?

Reply:
I changed the name to ClientRegistry. I will respectfully insist on seperation of these modules for the reasons like clarity of purposes and SRP. But if it really is giving you an itch I can merge them :)